packages/login/src/components/passwordUpdate/PasswordUpdateDialog/PasswordInput.js

```diff
@@ -1,30 +1,36 @@
 import PropTypes from 'prop-types'
 import React from 'react'
-import classNames from 'classnames'
+import {design, StatedValue} from 'tocco-ui'
+
+import {StyledPasswordInput, StyledPasswordInputWrapper} from './StyledPasswordInput'
 
 const PasswordInput = props => {
-  const className = classNames(
-    props.name,
-    'form-group',
-    {'has-success': props.value && props.valid === true},
-    {'has-error': props.value && props.valid === false}
-  )
+  const getSignal = props.value
+    ? props.valid
+      ? design.condition.SUCCESS
+      : design.condition.DANGER
+    : null
 
   return (
-    <div className={className}>
-      <label htmlFor={`${props.name} Input`}>{props.label}</label>
-      <input
-        type="password"
-        className="form-control"
-        id={`${props.name} Input`}
-        value={props.value}
-        onChange={e => props.onChange(e.target.value)}
-        readOnly={props.readOnly === true}
-        disabled={props.readOnly === true}
-        onKeyDown={props.onKeyDown}
-        autoFocus={props.autoFocus}
-      />
-    </div>
+    <StatedValue
+      hasValue={!!props.value}
+      id={`${props.name}-input`}
+      immutable={props.readOnly === true}
+      label={props.label}
+      signal={getSignal}
+    >
+      <StyledPasswordInputWrapper immutable={props.readOnly === true}>
+        <StyledPasswordInput
+          id={`${props.name}-input`}
```

Comment:
id attributes used this way are prone to lead to conflicts and, thus, to problems.

can we, somehow, get rid of them?

- are they really necessary? isn't it possible to change the markup in StatedValue so that the input component is nested in the label component?
- if they are necessary, could we choose a slightly different approach where StatedValue generates a unique id which it passes down to the input component?

Reply:
I totally agree. From React perspective there is no need for an id attribute. Probably a jQuery plugin or Tocco Framework depend on id's. Before investing it I'd like to ask @MethodenMann if he remember why id's were implemented. There are some other places in tocco-client where id's are used.